File: srm198/reppity.py
class Reppity:
	def longestRep(self, input):
		res = 0
		if len(input) == 2 and input[0] == input[1]: return 1
		for i in range(len(input)):
			for j in range(i+1, len(input)):
				tmp = input[i:j]
				if input[j:].find(tmp) != -1: res = max(res, len(tmp))
		return res

File: srm198/Reppity.java
import java.lang.*;

public class Reppity {

	public int longestRep(String input) {
		int res = 0;
		for (int i = 0; i < input.length(); i++) {
			for (int j = i + 1; j < input.length(); j++) {
				String tmp = input.substring(i, j);
				String tmp2 = input.substring(j);
				if (tmp2.indexOf(tmp) != -1) {
					res = Math.max(res, tmp.length());
				}
			}
		}
	return res;
	}
}